src/bridge/cockpitrouter.c

```diff
@@ -785,7 +819,7 @@ cockpit_router_init (CockpitRouter *self)
 
   /* Owns the channels */
   self->channels = g_hash_table_new_full (g_str_hash, g_str_equal, NULL, object_unref_if_not_null);
-  self->groups = g_hash_table_new_full (g_str_hash, g_str_equal, g_free, g_free);
+  self->groups = g_hash_table_new_full (g_str_hash, g_str_equal, g_free, channel_group_unref);
   self->fences = g_hash_table_new_full (g_str_hash, g_str_equal, g_free, NULL);
 
   /* The rules, including a default */
```

Comment:
Honestly, I'm not sure how testing this would work. I can write a test that opens a couple of channels with different groups, but how do I find out that they're actually using different D-Bus connections?